src/validation.cpp

```diff
@@ -1732,8 +1732,7 @@ bool CheckInputScripts(const CTransaction& tx, TxValidationState& state,
         // Verify signature
         CScriptCheck check(txdata.m_spent_outputs[i], tx, i, flags, cacheSigStore, &txdata);
         if (pvChecks) {
-            pvChecks->push_back(CScriptCheck());
-            check.swap(pvChecks->back());
+            pvChecks->push_back(std::move(check));
```

Comment:
Review note in 78030e0c08ce8ab30bde05ce97def571f05f1399: Calling the `swap` member function ensured at compile time that the object is not `const`. Now it is done at compile time only with clang-tidy `performance-move-const-arg`?

Reply:
[Updated](https://github.com/bitcoin/bitcoin/pull/26749#issuecomment-1370149202). Move semantics forced by making `CScriptCheck` a non-copyable class.

Reply:
In commit "consensus, refactor: Avoid `CScriptCheck::swap` in `CheckInputScripts`" (97292dafb528d2a097f29e4c79a08cced49e5451)

Should this use emplace_back instead of push_back?

Reply:
> > Should this use emplace_back instead of push_back?
> 
> I didn't mention this because I assumed the two functions would do the same if they received a moved object that is already fully constructed?

Oh, I think you are right they are equivalent in this case. I still think it would be a little better to use emplace_back for readability though, because with emplace_back you can be sure object is being constructed directly, while with push_back you have to check the types to know if a temporary is being created because of an implicit conversion.

Reply:
Thanks! [Updated](https://github.com/bitcoin/bitcoin/pull/26749#issuecomment-1472282805).